### PR TITLE
Jovial-Developer user davis6jm in class code review commit

### PR DESCRIPTION
**Pull request by Jovial-Developer**
Description of the project:
This is an app that is a simple do it yourself receipt library for people to learn to cook quick meals at home.

What I learned from this code review.
* I learned how to integrate a new field in a DTO not just create the field, but apply it in the project
* I learned to use an interface enforcing Dependency Inversion principle in SOLID
* I learned how to fix integrations tests to test for the newly created field.
* I learned how to adjust the stubs correctly so the integration tests pass as well.

This Project was available on GitHub on time, This program did build. This program did compile. This program does run on an emulator/physical phone Program does not have many comments, but variables are descriptive enough to understand the code. This project follows Kotlin coding conventions.  More will descriptions will be added in pull request.

### CODE REVIEW

**Comment by Jovial-Developer** [comment]
Previous commits I made in the first sprint of the project I'm working on.
https://github.com/Jovial-Developer/IT3048CSpringProject/commit/92b3cfaa55318b8ee81f39824171beb9e6b038ce

https://github.com/Jovial-Developer/IT3048CSpringProject/commit/ecfbe9792ddfa8bf937e96eb1d63095e2671d1e7

https://github.com/Jovial-Developer/IT3048CSpringProject/commit/08dd9175b823757c8c0fb6303d526412eb5c6821

https://github.com/Jovial-Developer/IT3048CSpringProject/commit/e901359ea46a70a23df98e5bfedca99d4b59cfa7

**Comment by ccbrowndev** [comment]
Got changes for RecipeService.kt, closing

File: app/src/main/java/com/cookit/dto/Recipe.kt [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-    var instructions : String = "",
-    var category : String = "",
-    var cuisine : String = "",
-    var ingredients : MutableMap<String, String> = LinkedHashMap<String, String>()
+    @SerializedName ("Name") var name : String,
+    @SerializedName ("Instructions")var instructions : String = "",
+    @SerializedName ("Category")var category : String = "",
+    @SerializedName ("Cuisine")var cuisine : String = "",
+    @SerializedName ("possibleAllergens")var possibleAllergens : String = "",
+    @SerializedName ("Ingredients")var ingredients : MutableMap<String, String> = LinkedHashMap<String, String>()
```
Comment (discospiff) [inline app/src/main/java/com/cookit/dto/Recipe.kt:11]: Do the @SerializedName parameters match a JSON feed, with capitalization?  They should.

File: app/src/main/java/com/cookit/service/RecipeService.kt [hunk 2]
```diff
@@ -8,9 +8,12 @@ import kotlinx.coroutines.async
 import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
+interface IRecipeService {
+    suspend fun fetchRecipes() : Set<Recipe>?
+}
```
Comment (discospiff) [inline app/src/main/java/com/cookit/service/RecipeService.kt:13]: Good to use interfaces.  That aligns with SOLID.

File: app/src/test/java/com/cookit/RecipeServiceIntegrationTest.kt [hunk 3]
```diff
@@ -101,6 +115,18 @@ class RecipeServiceIntegrationTest {
         assertTrue(containsPBJ)
     }
 
+    private fun thenRecipeCollectionHasPossibleAllergenFieldReturningPeanuts() {
+        assertNotNull(allRecipes)
+        assertTrue(allRecipes!!.isNotEmpty())
+        var containsAllergen = false
+        allRecipes!!.forEach {
```
Comment (discospiff) [inline app/src/test/java/com/cookit/RecipeServiceIntegrationTest.kt:122]: Use ?.let{} to handle nullness, instead of force unwrap (!!)

**Comment by discospiff** [comment]
It looks like the group has already closed this, though I'd see what can be harvested.  More unit test coverage is always a good idea.

**Pull request closed without merging**